DSCResources/MSFT_xDNSServerAddress/MSFT_xDNSServerAddress.psm1

```diff
@@ -47,36 +55,46 @@ function Get-TargetResource
     Write-Verbose -Message ( @( "$($MyInvocation.MyCommand): "
         $($LocalizedData.GettingDNSServerAddressesMessage)
         ) -join '')
+    
+    # obtain all the interface addresses
+    $dnsClientServerAddresses = Get-NetAdapter -Name $InterfaceAlias `
+    | ForEach-Object { Get-DnsClientServerAddress -InterfaceAlias $_  -AddressFamily $AddressFamily }
 
     $returnValue = @{
-        Address = (Get-DnsClientServerAddress `
-            -InterfaceAlias $InterfaceAlias `
-            -AddressFamily $AddressFamily).ServerAddresses
+        Address = $dnsClientServerAddresses.ServerAddresses
         AddressFamily = $AddressFamily
         InterfaceAlias = $InterfaceAlias
     }
 
-    $returnValue
+    return $returnValue
 }
 
-######################################################################################
-# The Set-TargetResource cmdlet.
-# This function will set a new Server Address in the current node
-######################################################################################
+#endregion
+
+#region Set-TargetResource function
+
 function Set-TargetResource
 {
+<#
```

Comment:
Could we modify this to:
````powershell
function Test-TargetResource
{
    <#
        .SYNOPSIS 
        Test Node configuration is in desired state or not.

        .DESCRIPTION
        Test if the given Server Address is among the current node's Server Address collection.
    #>
````

That way it collapses nicely in most code folding editors.